myFitnessTrainer/src/screens/AvatarScreen.js

```diff
@@ -0,0 +1,127 @@
+  const [defaultImageURL, setDefaultImageURL] = useState(null)
+  const [avatarURL, setAvatarURL] = useState(null)
+  const [image, setImage] = useState(null);
+
+  // get the default.png from Firebase. We'll use its URL as the User's Avatar URL if
+  // custom image is not picked from the user's device
+  const defaultImageRef = ref(storage, defaultAvatarFileName);
+  getDownloadURL(defaultImageRef)
+    .then((url) => {
+      setDefaultImageURL(url);
+      setAvatarURL(url);
+    });
+
+  const pickImage = async () => {
+    let result = await ImagePicker.launchImageLibraryAsync({
+      mediaTypes: ImagePicker.MediaTypeOptions.Images,
+      allowsEditing: true,
+      aspect: [4, 3],
+      quality: 0,
+    });
+    if (!result.canceled) {
+      setImage(result.assets[0].uri);
+    }
+  };
+
+  const saveAvatarSelection = async () => {
+    if (image){ // if user picked an image from their device
+      // upload image to Firebase
+      // https://github.com/expo/examples/blob/master/with-firebase-storage-upload/App.js#L178-L205
+      const blob = await new Promise((resolve, reject) => {
+        const xhr = new XMLHttpRequest();
+        xhr.onload = function () {
+          resolve(xhr.response);
+        };
+        xhr.onerror = function (e) {
+          console.log(e);
+          reject(new TypeError("Network request failed"));
+        };
+        xhr.responseType = "blob";
+        xhr.open("GET", image, true);
+        xhr.send(null);
+      });
+      // save image from device as "avatar.png"
+      const fileRef = ref(getStorage(), pickedImageFileName);
+      await uploadBytes(fileRef, blob);
+      blob.close();
+
+      // get URL of avatar.png
+      const pickedImageRef = ref(storage, pickedImageFileName);
+      getDownloadURL(pickedImageRef)
+        .then((url) => {
+          // set user's avatar URL as the URL of avatar.png
+          setAvatarURL(url);
+        });
+      console.log("image was picked; avatarURL=", avatarURL)
+    
+    let newStorageLocation = "gs://" + pickedImageRef._location.bucket + "/" + pickedImageFileName;
+    // Ex: newStorageLocation = gs://myfitnesstrainer-94289.appspot.com/avatar.png
+    console.log("newStorageLocation=", newStorageLocation);
+    // TODO: implement functionality to update the User's avatars/avatar storageLocation 
+    } 
+    else {
+      // TODO: this section is mainly for debugging. Erase this else section when we implement 
+      // functionality to update the User's avatar image URL attribute
+
+      console.log("no image was picked; using default. avatarURL=", avatarURL);
+
+      // TODO: in theory, if we don't pick a new image, the user's avatars/avatar storage location 
+      // will remain gs://myfitnesstrainer-94289.appspot.com/default.png. However, this value will 
+      // probably get altered in testing/developing the app, so we can perhaps look into setting it
+      // here, or when we define const defaultImageRef
```

Comment:
I'm ok with having the default user's avatar point to the same location at gs://myfitnesstrainer-94289.appspot.com/default.png

Reply:
Altered the comment / TODO a bit and moved it to line 23 near where we define const defaultImageRef
The TODO will be a good `UI - AvatarScreen 2nd Iteration` task. Let me know if I misinterpreted your comment!